templates/_macros/vf_cta-section.jinja

```diff
@@ -0,0 +1,105 @@
+# All Params
+  # title_text: H2 title text
+  # link_text: H2 Link text
+  # link_href: href for the link text
+  # layout: Layout type of cta section. Options are "cta-default-100", "cta-default-25-75", "cta-block-100", "cta-block-25-75". Default is "cta-default-100".
+  # has_chevron: Whether the link text should have a chevron icon. Default is True.
+
+# All Slots
+  # title: Optional title slot. Only applicable to "cta-default-100" and "cta-default-25-75" layouts. When specified, the "title_text" parameter will be ignored.
+  # description: Paragraph-style (one or more) content below the title. This slot is required for "cta-block-100" and "cta-block-25-75" layouts.
+  # cta: Call-to-action block with action links/buttons. This slot is required for "cta-block-100" and "cta-block-25-75" layouts.
+
+# Variants:
+  # cta-default-100: Full-width CTA with title and link text
+    # Params:
+        # title_text: H2 title text - optional
+        # link_text: H2 link text - required
+        # link_href: href for the h2 link - required
+        # layout: the variant - cta-default-100
+        # has_chevron: Whether the link text should have a chevron icon - Optional, default is True
+
+    # Slots:
+        # title: When specified, the "title_text" parameter will be ignored - Optional
+
+  # cta-default-25-75: 25-75 percent width CTA with title and link text
+    # Params:
+        # title_text: H2 title text - optional
+        # link_text: H2 link text - required
+        # link_href: href for the h2 link - required
+        # layout: the variant - cta-default-25-75
+        # has_chevron: Whether the link text should have a chevron icon - Optional, default is True
+
+      # Slots:
+        # title: When specified, the "title_text" parameter will be ignored - Optional
+
+  # cta-block-100: Full width CTA with title, description and cta block with action links
+    # Params:
+        # title_text: H2 title text - required
+        # layout: the variant - cta-block-100
+
+    # Slots:
+        # description: Paragraph-style (one or more) content below the title - Optional
+        # cta: Call-to-action block (required)
+
+  # cta-block-25-75: 25-75 percent width CTA with title, description and cta block with action links
+    # Params:
+        # title_text: H2 title text - required
+        # layout: the variant - cta-block-25-75
+
+    # Slots:
+        # description: Paragraph-style (one or more) content below the title - Optional
+        # cta: Call-to-action block (required)
+
+{% macro vf_cta_section(title_text, link_text, link_href, has_chevron=True, layout='cta-default-100', caller=None) %}
```

Comment:
A bit of a nitpick as most users of these patterns are consuming Jinja, not HTML: your macro is generating a lot of whitespace in the outputted markup. Switch the pattern examples to `HTML` mode to see their output.

![image](https://github.com/user-attachments/assets/db510466-2f3d-4118-971a-df94ac1d9744)

Could you go through and adjust the [whitespace control characters](https://jinja.palletsprojects.com/en/stable/templates/#whitespace-control) in the macro so that there is not excessive whitespace in the markup?